File: quan.java/message/src/test/java/quan/message/role/RoleInfo.java
package quan.message.role;

import java.util.*;
import java.io.IOException;
import quan.message.*;

/**
 * 角色信息<br/>
 * 自动生成
 */
public class RoleInfo extends Bean {

    //角色id
    private long id;

    //角色名
    private String roleName = "";

    private RoleType roleType;

    private boolean b;

    private short s;

    private int i;

    private float f;

    private double d;

    private byte[] data = new byte[0];

    private ArrayList<Integer> list = new ArrayList<>();

    private HashSet<Integer> set = new HashSet<>();

    private HashMap<Integer, Integer> map = new HashMap<>();

    public RoleInfo() {
    }

    /**
     * 角色id
     */
    public long getId() {
        return id;
    }

    /**
     * 角色id
     */
    public RoleInfo setId(long id) {
        this.id = id;
        return this;
    }

    /**
     * 角色名
     */
    public String getRoleName() {
        return roleName;
    }

    /**
     * 角色名
     */
    public RoleInfo setRoleName(String roleName) {
        Objects.requireNonNull(roleName);
        this.roleName = roleName;
        return this;
    }

    public RoleType getRoleType() {
        return roleType;
    }

    public RoleInfo setRoleType(RoleType roleType) {
        this.roleType = roleType;
        return this;
    }

    public boolean getB() {
        return b;
    }

    public RoleInfo setB(boolean b) {
        this.b = b;
        return this;
    }

    public short getS() {
        return s;
    }

    public RoleInfo setS(short s) {
        this.s = s;
        return this;
    }

    public int getI() {
        return i;
    }

    public RoleInfo setI(int i) {
        this.i = i;
        return this;
    }

    public float getF() {
        return f;
    }

    public RoleInfo setF(float f) {
        this.f = f;
        return this;
    }

    public double getD() {
        return d;
    }

    public RoleInfo setD(double d) {
        this.d = d;
        return this;
    }

    public byte[] getData() {
        return data;
    }

    public RoleInfo setData(byte[] data) {
        Objects.requireNonNull(data);
        this.data = data;
        return this;
    }

    public ArrayList<Integer> getList() {
        return list;
    }

    public HashSet<Integer> getSet() {
        return set;
    }

    public HashMap<Integer, Integer> getMap() {
        return map;
    }

    @Override
    public void encode(Buffer buffer) throws IOException {
        super.encode(buffer);

        buffer.writeLong(this.id);
        buffer.writeString(this.roleName);
        buffer.writeInt(this.roleType == null ? 0 : this.roleType.getValue());
        buffer.writeBool(this.b);
        buffer.writeShort(this.s);
        buffer.writeInt(this.i);
        buffer.writeFloat(this.f, 2);
        buffer.writeDouble(this.d);
        buffer.writeBytes(this.data);

        buffer.writeInt(this.list.size());
        for (int $list$Value : this.list) {
            buffer.writeInt($list$Value);
        }

        buffer.writeInt(this.set.size());
        for (int $set$Value : this.set) {
            buffer.writeInt($set$Value);
        }

        buffer.writeInt(this.map.size());
        for (int $map$Key : this.map.keySet()) {
            buffer.writeInt($map$Key);
            buffer.writeInt(this.map.get($map$Key));
        }
    }

    @Override
    public void decode(Buffer buffer) throws IOException {
        super.decode(buffer);

        this.id = buffer.readLong();
        this.roleName = buffer.readString();
        this.roleType = RoleType.valueOf(buffer.readInt());
        this.b = buffer.readBool();
        this.s = buffer.readShort();
        this.i = buffer.readInt();
        this.f = buffer.readFloat(2);
        this.d = buffer.readDouble();
        this.data = buffer.readBytes();

        int $list$Size = buffer.readInt();
        for (int i = 0; i < $list$Size; i++) {
            this.list.add(buffer.readInt());
        }

        int $set$Size = buffer.readInt();
        for (int i = 0; i < $set$Size; i++) {
            this.set.add(buffer.readInt());
        }

        int $map$Size = buffer.readInt();
        for (int i = 0; i < $map$Size; i++) {
            this.map.put(buffer.readInt(), buffer.readInt());
        }
    }

    @Override
    public String toString() {
        return "RoleInfo{" +
                "id=" + id +
                ",roleName='" + roleName + '\'' +
                ",roleType=" + roleType +
                ",b=" + b +
                ",s=" + s +
                ",i=" + i +
                ",f=" + f +
                ",d=" + d +
                ",data=" + Arrays.toString(data) +
                ",list=" + list +
                ",set=" + set +
                ",map=" + map +
                '}';

    }

}

File: quan.lua/test/src/test/message/role/RoleInfo.lua
local Buffer = require("quan.message.Buffer")
local Message = require("quan.message.Message")

---
---角色信息
---自动生成
---
local RoleInfo = {
    ---类名
    class = "test.message.role.RoleInfo",
}

local function onSet(table, key, value)
    assert(not RoleInfo[key], "不允许修改只读属性:" .. key)
    rawset(table, key, value)
end

---
---角色信息.构造
---@param args 参数列表可以为空
---
function RoleInfo.new(args)
    assert(args == nil or type(args) == "table", "参数错误")
    args = args or {}

    local instance = {
        ---角色id
        id = args.id or 0,
        ---角色名
        name = args.name or "",
        type = args.type or 0,
        b = args.b or false,
        s = args.s or 0,
        i = args.i or 0,
        f = args.f or 0.0,
        d = args.d or 0.0,
        data = args.data or "",
        list = args.list or {},
        set = args.set or {},
        map = args.map or {},
    }

    instance = setmetatable(instance, { __index = RoleInfo, __newindex = onSet })
    return instance
end

---
---角色信息.编码
---@param buffer quan.message.Buffer 可以为空
---@return quan.message.Buffer
---
function RoleInfo:encode(buffer)
    assert(type(self) == "table" and self.class == RoleInfo.class, "参数[self]类型错误")
    assert(buffer == nil or type(buffer) == "table" and buffer.class == Buffer.class, "参数[buffer]类型错误")
    buffer = buffer or Buffer.new()

    buffer:writeLong(self.id)
    buffer:writeString(self.name)
    buffer:writeInt(self.type or 0);
    buffer:writeBool(self.b)
    buffer:writeShort(self.s)
    buffer:writeInt(self.i)
    buffer:writeFloat(self.f, 2)
    buffer:writeDouble(self.d)
    buffer:writeBytes(self.data)

    buffer:writeInt(#self.list)
    for i, value in ipairs(self.list) do
        buffer:writeInt(value);
    end

    buffer:writeInt(#self.set)
    for i, value in ipairs(self.set) do
        buffer:writeInt(value);
    end

    buffer:writeInt(table.size(self.map))
    for key, value in pairs(self.map) do
        buffer:writeInt(key)
        buffer:writeInt(value)
    end

    return buffer
end

---
---角色信息.解码
---@param buffer quan.message.Buffer 不能为空
---@param self test.message.role.RoleInfo 可以为空
---@return test.message.role.RoleInfo
---
function RoleInfo.decode(buffer, self)
    assert(type(buffer) == "table" and buffer.class == Buffer.class, "参数[buffer]类型错误")
    assert(self == nil or type(self) == "table" and self.class == RoleInfo.class, "参数[self]类型错误")
    self = self or RoleInfo.new()

    self.id = buffer:readLong()
    self.name = buffer:readString()
    self.type = buffer:readInt();
    self.b = buffer:readBool()
    self.s = buffer:readShort()
    self.i = buffer:readInt()
    self.f = buffer:readFloat(2)
    self.d = buffer:readDouble()
    self.data = buffer:readBytes()

    for i = 1, buffer:readInt() do
        self.list[i] = buffer:readInt()
    end

    for i = 1, buffer:readInt() do
        self.set[i] = buffer:readInt()
    end

    for i = 1, buffer:readInt() do
        self.map[buffer:readInt()] = buffer:readInt()
    end

    return self
end

RoleInfo = table.readOnly(RoleInfo)
return RoleInfo